src/gameplayv2/resistance/components/RoleComponent.js

```diff
@@ -0,0 +1,7 @@
+//extends component
+//define a class for role component
+export class RoleComponent{
```

Comment:
A Role isn't a component either. A role is an entity that is made up of particular components. 

E.g. Assassin is a role, and is the entity that contains the following components: "CanVote", "CanPick", "CanShootMerlin", "CanSeeSpies". Etc.